pallas-network/src/facades.rs

```diff
@@ -191,6 +191,43 @@ impl NodeClient {
         })
     }
 
+    #[cfg(target_os = "windows")]    
+    pub async fn connect(pipe_name: impl AsRef<std::ffi::OsStr>, magic: u64) -> Result<Self, Error> {
+        debug!("connecting");
+
+        let bearer = Bearer::connect_named_pipe(pipe_name)
+            .await
+            .map_err(Error::ConnectFailure)?;
+
+        let mut plexer = multiplexer::Plexer::new(bearer);
+
+        let hs_channel = plexer.subscribe_client(PROTOCOL_N2C_HANDSHAKE);
+        let cs_channel = plexer.subscribe_client(PROTOCOL_N2C_CHAIN_SYNC);
+        let sq_channel = plexer.subscribe_client(PROTOCOL_N2C_STATE_QUERY);
+
+        let plexer_handle = tokio::spawn(async move { plexer.run().await });
+
+        let versions = handshake::n2c::VersionTable::only_v10(magic);
+        let mut client = handshake::Client::new(hs_channel);
+
+        let handshake = client
+            .handshake(versions)
+            .await
+            .map_err(Error::HandshakeProtocol)?;
+
+        if let handshake::Confirmation::Rejected(reason) = handshake {
+            error!(?reason, "handshake refused");
+            return Err(Error::IncompatibleVersion);
+        }
+
+        Ok(Self {
+            plexer_handle,
+            handshake,
+            chainsync: chainsync::Client::new(cs_channel),
+            statequery: localstate::Client::new(sq_channel),
+        })
```

Comment:
can we move this block into it's own platform-agnostic private function called `connect_bearer(bearer, magic)`?

Just to avoid repetition of logic that is shared between platforms.

Reply:
Created now but with this sig:
```
async fn connect_bearer(bearer:Bearer,versions: VersionTable<VersionData>) -> Result<Self, Error>  {...}
```

I am not sure if it is due to a bug or working as designed , but it is only possible to connect to the pipe using only_v10 - otherwise we get back this type of error: 

`ERROR pallas_network::facades: handshake refused, reason: HandshakeDecodeError(32783, "unknown encoding: TList [TInt 1,TBool True]")`